packages/core/lib/debug/printer.js

```diff
@@ -138,12 +159,14 @@ class DebugPrinter {
       return;
     }
 
-    const source = this.session.view(solidity.info.sources)[compilationId].byId[
-      sourceId
-    ].source;
-    //we don't just get extract this from the location because passed-in location may be
-    //missing the soure text
-    const colorizedSource = this.colorizedSources[compilationId][sourceId];
+    const { source, internal, internalFor } = this.session.view(
+      solidity.current.sources
+    )[sourceId];
+    //we don't just get extract the source text from the location because passed-in location may be
+    //missing the source text
+    const colorizedSource = internal
+      ? this.colorizedSources[compilationId].internal[internalFor][sourceId]
+      : this.colorizedSources[compilationId].user[sourceId];
```

Comment:
Is there a way to factor this so we can avoid distinguishing here about internalness?

Reply:
There sure is :)